packages/router/addon/index.js

```diff
@@ -61,56 +60,39 @@ if (macroCondition(getGlobalConfig()['@embroider/core']?.active)) {
     // overriding to provide our own handlerResolver.
     setupRouter() {
       let isSetup = this._super(...arguments);
+
+      // eslint-disable-next-line ember/no-private-routing-service
+      const routerMicrolib = this._routerMicrolib;
```

Comment:
I've left the original "private API" usage because:

1) I didn't want to change to much all at once
2) I'm not 100% sure yet if the new APIs work across the needed support versions (this is likely something a single follow up PR should focus on ... assuming is is possible)

Reply:
Ya, sounds good to me. Maybe make an issue to track that?

Reply:
Ya, I plan to continue working on that category of work until embroider code-base is as modern as it can be.